File: CryptographicBackend/src/main/java/cl/niclabs/cb/backend/methods/SignInitMethod.java
package cl.niclabs.cb.backend.methods;

import cl.niclabs.cb.backend.Method;

public interface SignInitMethod extends Method {
	public static class Args {
        public final String sessionHandler;
		public final String mechanism;
		public final String keyHandler;

        public Args(String sessionHandler, String mechanism, String keyHandler) {
            this.sessionHandler = sessionHandler;
            this.mechanism = mechanism;
            this.keyHandler = keyHandler;
        }
    }
}

File: CryptographicFrontend/communication/SignInitMethod.cpp
#include "SignInitMethod.h"
#include "ResponseMessage.h"
#include "Argument.h"

using namespace communication;

SignInitMethod::SignInitMethod ( std::string sessionHandler, std::string mechanism, std::string privateKeyHandler )
    : Method ( "SignInit" )
{
    addArgument ( "sessionHandler", sessionHandler );
    addArgument ( "mechanism", mechanism );
    addArgument ( "keyHandler", privateKeyHandler );
}

ResponseMessage SignInitMethod::parseResponse ( const Json::Value& value )
{
    return ResponseMessage();
}
// kate: indent-mode cstyle; replace-tabs on; 
